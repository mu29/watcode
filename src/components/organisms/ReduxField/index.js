import React from 'react'
import { PropTypes } from 'prop-types'
import { Field } from 'components'

const ReduxField = ({
  input, meta, ...props
}) => {
  const fieldProps = {
    ...input,
    ...meta,
    ...props,
    error: meta.touched && meta.error ? meta.error : '',
  }

  return (
    <Field { ...fieldProps } />
  )
}

ReduxField.propTypes = {
  input: PropTypes.shape({
    name: PropTypes.string,
    value: PropTypes.string,
    onChange: PropTypes.func,
  }).isRequired,
  meta: PropTypes.shape({
    touched: PropTypes.bool,
    error: PropTypes.string,
  }).isRequired,
}

export default ReduxField
