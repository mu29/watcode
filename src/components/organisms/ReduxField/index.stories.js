import React from 'react'
import { storiesOf } from '@storybook/react'
import { ReduxField } from 'components'

storiesOf('Organisms/ReduxField', module)
  .add('기본', () => (
    <ReduxField input={{ name: 'name' }} meta={{}} />
  ))
  .add('라벨', () => (
    <ReduxField input={{ name: 'name' }} label="이메일" meta={{}} />
  ))
  .add('오류', () => (
    <ReduxField input={{ name: 'name' }} label="이메일" meta={{ touched: true, error: '이메일 주소를 확인해주세요' }} />
  ))
