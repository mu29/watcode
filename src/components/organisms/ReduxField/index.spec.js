import React from 'react'
import { shallow } from 'enzyme'
import ReduxField from '.'

let meta
let input
const wrap = (props = {}) => shallow(<ReduxField { ...{ meta, input, ...props } } />)

beforeEach(() => {
  meta = {
    touched: false,
    error: null,
  }
  input = {
    name: 'test',
  }
})

describe('<ReduxField />', () => {
  it('임의의 속성을 그려준다', () => {
    const wrapper = wrap({ id: 'foo' })
    expect(wrapper.find({ id: 'foo' })).toHaveLength(1)
  })

  it('touched 되지 않았다면 오류를 표시하지 않는다', () => {
    const wrapper = wrap()
    expect(wrapper.find({ error: 'test' }).length).toBe(0)
    wrapper.setProps({ meta: { error: 'test' } })
    expect(wrapper.find({ error: 'test' }).length).toBe(0)
  })

  it('touched 되었고 오류가 있는 경우에 해당 오류를 표시한다', () => {
    const wrapper = wrap({ meta: { touched: true, error: 'test' } })
    expect(wrapper.find({ error: 'test' })).toHaveLength(1)
  })
})
