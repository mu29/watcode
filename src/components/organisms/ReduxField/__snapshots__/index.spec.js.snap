// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReduxField /> 정상적으로 렌더링된다 1`] = `
<InputField
  error=""
  name="test"
  touched={false}
/>
`;
